shared/android/app/src/main/java/io/keybase/ossifrage/modules/NativeLogger.java

```diff
@@ -14,69 +13,46 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
-import java.io.StringWriter;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 
 public class NativeLogger extends ReactContextBaseJavaModule {
     private static final String NAME = "KBNativeLogger";
+    private static final String RN_NAME = "ReactNativeJS";
 
-    // Must match strings passed to NativeLogger in
-    // shared/logger/index.js.
-    private static final String ERROR_TAG = "e";
-    private static final String INFO_TAG = "i";
-    private static final String WARN_TAG = "w";
 
     private static void rawLog(String tag, String jsonLog) {
         Log.i(tag + NAME, jsonLog);
     }
 
-    // This should do roughly the same thing as dumpLine from
-    // native-logger.js.
-    private static String dumpLine(String toLog) {
-        long millis = System.currentTimeMillis();
-        StringWriter sw = new StringWriter();
-        JsonWriter js = new JsonWriter(sw);
-        try {
-            js.beginArray()
-                .value(millis)
-                .value(toLog)
-                .endArray()
-                .close();
-            return sw.toString();
-        } catch (IOException e) {
-            rawLog(ERROR_TAG, "Exception in dumpLine: " + Log.getStackTraceString(e));
-            return toLog;
-        }
-    }
-
-    private static String dumpLine(String toLog, Throwable tr) {
-      return dumpLine(toLog + ": " + Log.getStackTraceString(tr));
+    private static String formatLine(String tagPrefix, String toLog) {
+        // Copies the Style JS outputs in native/logger.native.tsx
+        return tagPrefix + NAME + ": [" + System.currentTimeMillis() + ",\"" + toLog + "\"]";
```

Comment:
Any reason this doesn't use `String.format`?

Reply:
Yeah I'm pretty sure String.format is like 10x slower than a simple string concat.